kanvas/src/jsMain/kotlin/HtmlCanvas.kt

```diff
@@ -53,21 +51,17 @@ public class HtmlCanvas(
         val startAngleRadians = startAngle * PI / 180f
         val endAngleRadians = (startAngle + sweepAngle) * PI / 180f
         context.ellipse(x, y, radiusX, radiusY, rotation = 0.0, startAngleRadians, endAngleRadians)
-        when (paint) {
-            is Paint.Stroke -> context.stroke()
-            else -> context.fill()
-        }
+        if (paint !is Paint.Stroke) context.fill()
+        if (paint !is Paint.Fill) context.stroke()
```

Comment:
I imagine there must be a good reason for the `!is` for these?
If so, would probably be good to have a small code comment explaining why `is` couldn't be used.

Notices that these two lines are repeated in other functions below, would `context.applyPaint(paint)` makes sense? If you do end up leaving a code comment in there, it would make it so the code comment only need to go in a single place rather than in all those functions. 🤷

Reply:
I was using `paint !is Paint.Stroke` to save a few keyboard clicks over `paint is Paint.Fill || paint is Paint.FillAndStroke`. I've extracted that out into some documented `shouldFill` and `shouldStroke` functions now.

I'm not a big fan of extracting the two checks into a function that handles it, because out of the 5 times I use that pattern, only 3 of them share an implementation (`fill()` and `stroke()` without arguments). Defining the function alone takes more than the 3 lines it would save, and that's before getting into the complexity of passing lambda functions to handle the other 2 cases.